### ISSUE TITLE
CSE 2.0 :  Couldn't initialize cluster

### ISSUE THREAD

**Issue by Vijendrasi** [vmware/container-service-extension#352]
CSE 2.0
CSE server OS -> Photon OS : 3.0
VCD version : 9.5

Do we need internet connection for initialization of cluster ?
-------------------------------------------------------------------------------------------------------
root@CSEVM200 [ /home/vmware ]# vcd cse template list
catalog    catalog_item                            description        is_default    name

CSE2.0     photon-custom-hw11-2.0-304b817-k8s      PhotonOS v2        True          photon-v2
                                                   Docker 17.06.0-9
                                                   Kubernetes 1.12.7
                                                   weave 2.3.0
CSE2.0     ubuntu-16.04-server-cloudimg-amd64-k8s  Ubuntu 16.04       False         ubuntu-16.04
                                                   Docker 18.06.3~ce
                                                   Kubernetes 1.13.5
                                                   weave 2.3.0

---------  --------------------------------------  -----------------  ------------  ------------
Photon Template :

vcd cse cluster create CSE-PH-CLU01 --network ORG-NET-EXT01 --nodes 1
create_cluster: Creating cluster CSE-PH-CLU01(97258e4a-f7dc-4108-a9b7-9ec72e79f4d5)
create_cluster: Creating cluster vApp CSE-PH-CLU01(97258e4a-f7dc-4108-a9b7-9ec72e79f4d5)
create_cluster: Creating master node for CSE-PH-CLU01(97258e4a-f7dc-4108-a9b7-9ec72e79f4d5)
create_cluster: Initializing cluster CSE-PH-CLU01(97258e4a-f7dc-4108-a9b7-9ec72e79f4d5)
task: de84f689-6357-4e7f-adf9-58b67d0e769d, result: error, message: Couldn't initialize cluster:
[preflight] Some fatal errors occurred:
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-apiserver:v1.12.7: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-controller-manager:v1.12.7: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-scheduler:v1.12.7: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-proxy:v1.12.7: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/pause:3.1: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/etcd:3.2.24: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/coredns:1.2.2: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
[preflight] If you know what you are doing, you can make a check non-fatal with...

---------  --------------------------------------  -----------------  ------------  ------------
Ubuntu template : 

 vcd cse cluster create CSE-UB-CLU01 --network ORG-NET-EXT01 --nodes 1 -t ubuntu-16.04
create_cluster: Creating cluster CSE-UB-CLU01(d9856103-afef-46ae-916a-33f4cdebe492)
create_cluster: Creating cluster vApp CSE-UB-CLU01(d9856103-afef-46ae-916a-33f4cdebe492)
create_cluster: Creating master node for CSE-UB-CLU01(d9856103-afef-46ae-916a-33f4cdebe492)
create_cluster: Initializing cluster CSE-UB-CLU01(d9856103-afef-46ae-916a-33f4cdebe492)
task: 7f1a46c3-fe3c-4f8a-92fe-9288dd42d99d, result: error, message: Couldn't initialize cluster:
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-apiserver:v1.13.5: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-controller-manager:v1.13.5: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-scheduler:v1.13.5: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-proxy:v1.13.5: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/pause:3.1: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/etcd:3.2.24: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/coredns:1.2.6: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
[preflight] If you know what you are doing, yo...

**Comment by sakthisunda** [comment]
Yes. Internet connection is required as part of the initialization process and executing customization scripts on the cluster, node creation etc.

**Comment by Vijendrasi** [comment]
I have used older version of CSE 1.2.5 that time internet connection was required for during template creation process. 
while running vcd cse cluster create CSE-UB-CLU01 --network ORG-NET-EXT01 --nodes 1 command internet connection was never required for ORG-NET-EXT01 network. I am sure it must be remain same in CSE 2.0 as well. It is not possible that all the workload vms will have internet connectivity.
